src/ripple/rpc/tests/LedgerData.test.cpp

```diff
@@ -0,0 +1,192 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/test/jtx.h>
+
+namespace ripple {
+
+class LedgerData_test : public beast::unit_test::suite
+{
+public:
+
+    static
+    std::unique_ptr<Config>
+    makeConfig(bool setup_admin)
+    {
+        auto p = std::make_unique<Config>();
+        test::setupConfigForUnitTests(*p);
+        // the default config has admin active
+        // ...we remove them if setup_admin is false
+        if (! setup_admin)
+        {
+            (*p)["port_rpc"].set("admin","");
+            (*p)["port_ws"].set("admin","");
+        }
+        return p;
+    }
+
+    // test helper
+    static bool checkArraySize(Json::Value const& val, unsigned int size)
+    {
+        return val.isArray() &&
+               val.size() == size;
+    }
+
+    // test helper
+    static bool checkMarker(Json::Value const& val)
+    {
+        return val.isMember(jss::marker) &&
+               val[jss::marker].isString() &&
+               val[jss::marker].asString().size() > 0;
+    }
+
+    void testCurrentLedgerToLimits(bool as_admin)
+    {
+        using namespace test::jtx;
+        Env env(*this, makeConfig(as_admin));
+        Account const gw ("gateway");
+        auto const USD  = gw["USD"];
+        env.fund(XRP(100000), gw);
+
+        const int max_limit = 256; //would be 2048 for binary requests, no need to test that here
+
+        for (auto i = 0; i < max_limit + 10; i++)
+        {
+            Account const bob (std::string("bob") + std::to_string(i));
```

Comment:
I proclaim henceforth that all constructor invocations shall use curly braces:
```
Account const bob{std::string("bob") + std::to_string(i)};
```
(except for `std::string`, standard containers, or any constructor that takes a `std::initializer_list`)